src/Microsoft.VisualStudio.Composition/Reflection/TypeRef.cs

```diff
@@ -242,7 +262,38 @@ public TypeRef MakeGenericTypeRef(ImmutableArray<TypeRef> genericTypeArguments)
 
             // We use the resolver parameter instead of the field here because this TypeRef instance
             // might have been constructed by TypeRef.Get(Type) and thus not have a resolver.
-            return new TypeRef(this.Resolver, this.AssemblyName, this.assemblyId, this.MetadataToken, this.FullName, this.TypeFlags, this.GenericTypeParameterCount, genericTypeArguments);
+            return new TypeRef(this.Resolver, this.AssemblyName, this.assemblyId, this.MetadataToken, this.FullName, this.TypeFlags, this.GenericTypeParameterCount, genericTypeArguments, this.BaseTypes, this.ElementTypeRef);
+        }
+
+        /// <summary>
+        /// Checks if the type represented by the given TypeRef can be assigned to the type represented by this instance.
+        /// The assignability check is done by traversing all the base types and interfaces of the given TypeRef to check
+        /// if any of them are equal to this instance. Should that fail, the CLR is asked to check for assignability
+        /// which will trigger an assembly load.
+        /// </summary>
```

Comment:
nit: move this to a `<remarks>` section. Since in effect the result is reliable, it's not super important to the caller of this method to know how it's done.